test/functional/sqlsrv/sqlsrv_AE_functions.inc

```diff
@@ -366,10 +367,15 @@ function testPatternMatch($conn, $tableName, $patterns, $dataTypes, $colNames, $
 
             foreach ($patternarray as $spattern) {
 
+                // Unicode operations with AE require the PHPTYPE to be specified to
+                // UTF-8 and the Latin1_General_BIN2 collation. If the COLLATE
+                // clause is left out, we get different results between the
+                // encrypted and non-encrypted columns (probably because the
+                // collation was only changed in the encryption query).
                 // We must pass the length of the pattern matching string
```

Comment:
For cases like pattern matching in sqlsrv, what happens if the length of pattern string is not passed in sqlsrv_prepare()?